app/main/posts/modify/post-media.directive.js

```diff
@@ -17,84 +19,44 @@ function (
         require: '^^form',
         scope: {
             mediaId: '=',
+            media: '=',
             name: '@'
         },
         templateUrl: 'templates/main/posts/modify/media.html',
         link: function ($scope, element, attr, formCtrl) {
-            // Initialize file container
-            $scope.fileContainer = {file: null};
+            $scope.showAdd = function () {
+                return (!$scope.media.id && !$scope.media.changed);
+            };
 
-            // Initialize media object
-            $scope.media = {};
+            $scope.showReplace = function () {
+                return $scope.media.dataURI || $scope.media.id;
+            };
+
+            $scope.showDelete = function () {
+                return $scope.media.id;
+            };
 
             if ($scope.mediaId) {
                 MediaEndpoint.get({id: $scope.mediaId}).$promise.then(function (media) {
                     $scope.media = media;
```

Comment:
I suspect this is likely to hit problems with the parent directive not picking this new object up..
You need to either 

1. pass the entire `medias` collection over the directive boundary, or...
2. make the post-edit directive a requirement for this one, pass its controller into this one and provide an interface there to add more media items